### PR TITLE
Fix cross-platform incompatibility with Storage Emulator exports

### PR DESCRIPTION
**Pull request by tohhsinpei**
### Description
This PR fixes https://github.com/firebase/firebase-tools/issues/4322#issuecomment-1084657623, hopefully the last follow up to #3647. The Storage emulator is unable to import data previously exported on a different platform because we didn't account for different file path formats (e.g. `\` delimiter on Windows but `/` on Linux) when flattening the directory structure.

Note re: the fix implemented here: ideally I would store and export everything in POSIX file format, then do the obvious replacement when importing. But I don't think that's a good idea for backwards compatibility reasons. Lmk if you have thoughts.

### Scenarios Tested
Added integration test to verify that emulator is able to import data previously exported on Windows.

### CODE REVIEW

File: src/emulator/storage/files.ts [hunk 1]
```diff
@@ -610,3 +613,11 @@ export class StorageLayer {
     }
   }
 }
+
+/** Returns file separator used in given path, either '\\' or '/'. */
+function getPathSep(decodedPath: string): string {
+  // Suffices to check first separator, which occurs immediately after bucket name
+  const bucketSuffix = ".appspot.com";
```
Comment (tonyjhuang) [inline src/emulator/storage/files.ts:620]: buckets don't have to end with .appspot.com. The guidelines for bucket naming state:

"""
Bucket names can only contain lowercase letters, numeric characters, dashes (-), underscores (_), and dots (.). Spaces are not allowed.
"""

See https://cloud.google.com/storage/docs/naming-buckets

Comment (tohhsinpei) [reply src/emulator/storage/files.ts]: Done.

File: scripts/storage-emulator-integration/import/windows-emulator-data/storage_export/metadata/fake-project-id.appspot.com%5Ctest_upload.jpg.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "test_upload.jpg",
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/import/windows-emulator-data/storage_export/metadata/fake-project-id.appspot.com%5Ctest_upload.jpg.json:2]: can you include a test with a nested directory structure

**Review by tonyjhuang — COMMENTED** [review]
thanks for the fix! LGTM outside of a test case

File: scripts/storage-emulator-integration/import/windows-emulator-data/storage_export/metadata/fake-project-id.appspot.com%5Ctest_upload.jpg.json (hunk 2 above)
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/import/windows-emulator-data/storage_export/metadata/fake-project-id.appspot.com%5Ctest_upload.jpg.json:2]: Done.

**Review by tonyjhuang — APPROVED** [review]

**Pull request merged**